src/net.cpp

```diff
@@ -3285,6 +3288,12 @@ bool CConnman::Start(CScheduler& scheduler, const Options& connOptions)
         scheduler.scheduleEvery([this] { ASMapHealthCheck(); }, ASMAP_HEALTH_CHECK_INTERVAL);
     }
 
+    auto reachable_nets{g_reachable_nets.All()};
+    const bool has_few_addresses{addrman.HasFewReachableAddresses(reachable_nets)};
+    if (has_few_addresses) {
+        networks_with_addrs = reachable_nets;
```

Comment:
So, to be explicit, this uses `g_reachable_nets` if `reachable_size < total_size / 2` otherwise all nets. So the switch is to avoid the cost of the filtering if the filtering doesn't much reduce the set size, yes? A comment to that effect might be helpful, as I had to puzzle out a bit why the value was set only in this case.

Reply:
> The switch is to avoid the cost of the filtering if the filtering doesn't much reduce the set size, yes?

Yes, I'll put a comment explaining that.